### PR TITLE
Implement Better Dynamic Slider

### PR DESCRIPTION
**Pull request by meponderR**
## Purpose
This changes the RAM slider to change depending on the amount of RAM the user has in their PC. With this change the Slider will go up to 4GB if the user has 4GB; 64 if the user has 64GB; etc. This is not hardcoded as opposed to #957 as requested by @killpowa in https://github.com/gorilla-devs/GDLauncher/issues/957#issuecomment-868727103. The original PR did not go through as the pull bot force pushed my branch.

## Approach
This uses a counter to cycle through 2048, 4096, 8192, 16384, etc

### CODE REVIEW

**Comment by PolpOnline** [comment]
This was addressed in 3a93317

**Comment by meponderR** [comment]
> This was addressed in [3a93317](https://github.com/gorilla-devs/GDLauncher/commit/3a93317087790ae86a30b528fec15c7ceb8d6e91)

This also adds a much better marking system than #957.

**Comment by Eskaan** [comment]
> > This was addressed in [3a93317](https://github.com/gorilla-devs/GDLauncher/commit/3a93317087790ae86a30b528fec15c7ceb8d6e91)
> 
> This also adds a much better marking system than #957.

No, it just sets a mark for every full number on the function x^2 and causes more non-needed cpu usage. We can do what you did simply by adding more marks on the slider.

**Comment by meponderR** [comment]
> No, it just sets a mark for every full mark on the function x^2 and causes more non-needed cpu usage. We can do what you did simply by adding more marks on the slider.

It was requested by killpowa as stated in the original comment. 

> This is not hardcoded as opposed to #957 as requested by @killpowa in [#957 (comment)](https://github.com/gorilla-devs/GDLauncher/pull/957#issuecomment-868727103)

Also, the cpu usage is still very low.

**Comment by Eskaan** [comment]
Yes, you’re right. But it still has to cycle through every `**` operation.
Whit too much hardcoding he is right but my point is: you just don’t need this function (btw the math behind it is great!)

**Comment by Eskaan** [comment]
Just look at my pr to fix this: #967
There was only one line to be changed and it works exactly the same way it does whit your version
Edit: I know 1000 is the wrong number, fix is also merged

**Comment by meponderR** [comment]
> Just look at my pr to fix this: #967
> There was only one line to be changed and it works exactly the same way it does whit your version
> Edit: I know 1000 is the wrong number, fix is also merged

The only point of this is if someone has 64, 128, 256, etc. gibibytes of ram and wants marks for the previous powers of 2.

**Comment by Eskaan** [comment]
If you really think someone needs markers that high, just add them lol

File: src/common/modals/Settings/components/Java.js [hunk 1]
```diff
@@ -152,8 +158,7 @@ export default function MyAccountPreferences() {
           `}
         >
           Disable this to specify a custom java path to use instead of using
-          openJDK shipped with GDLauncher if that is the case select the path to
-          your Java executable.
+          openJDK shipped with GDLauncher. Please select the java.exe binary
```
Comment (blarfoon) [inline src/common/modals/Settings/components/Java.js:161]: This won't work for linux/macOS

File: src/common/modals/Settings/components/Java.js [hunk 2]
```diff
@@ -98,13 +99,18 @@ function resetJavaArguments(dispatch) {
...
-  16384: '16384 MB',
-  32768: '32768 MB'
-};
+const maxMemorySlider = Math.round(totalmem() / 1024 / 1024 / 1024) * 1024;
+
+function getMarks() {
+  const totalMarks = {};
+  const totalMemory = Math.round(totalmem() / 1024 / 1024 / 1024);
+  const b2TotalMemory = Math.floor(Math.log2(totalMemory));
+  for (let b2Counter = b2TotalMemory; b2Counter > 0; b2Counter -= 1) {
```
Comment (blarfoon) [inline src/common/modals/Settings/components/Java.js:108]: Is it really necessary to go over every single number from `Math.log2(totalMemory)` to 0?
Looks like a lot of unnecessary loop steps, am I missing something?

File: src/common/modals/Settings/components/Java.js [hunk 3]
```diff
@@ -98,13 +99,18 @@ function resetJavaArguments(dispatch) {
...
-const marks = {
-  2048: '2048 MB',
-  4096: '4096 MB',
-  8192: '8192 MB',
-  16384: '16384 MB',
-  32768: '32768 MB'
-};
+const maxMemorySlider = Math.round(totalmem() / 1024 / 1024 / 1024) * 1024;
+
+function getMarks() {
```
Comment (blarfoon) [inline src/common/modals/Settings/components/Java.js:104]: This should eventually be extracted to an util function and imported into the 2 files

**Review by blarfoon — CHANGES_REQUESTED** [review]
This PR is good but there is still an issue with the number of marks. 
image.

I think a solution for this would be to count the vicinity of 2 marks and only add them to the marks array if they are far enough from each other.
Assuming we show marks for every power of 2GB
Let's say I have 8GB of RAM, that means marks will be 2, 4, 8GB. 
2-8 is 100% of the distance and 2-4 is 50% of it, we will show all of them.
Now let's say I have 32GB of RAM, that means we have marks for 2, 4, 8, 16, 32GB.
Now 2-32 is 100% of the distance but 2-4 is only 1/16th or around 6% of it. This is too low of a value and the text will overlap so we will not show it. I don't know the exact threshold but you can make a few tries. If you can address this change I will merge it!

**Comment by LilyRose2798** [comment]
Just an FYI for anyone looking at this PR, I have an open PR #1227 which addresses the issue mentioned here in a different way (rather than keeping a linear scale and changing where marks are placed, the scale is changed to be logarithmic so that all power of two GBs up to the user's memory can be displayed equally spaced without overlaps).

**Pull request closed without merging**